c/parallel/src/reduce.cu

```diff
@@ -98,8 +98,10 @@ reduce_runtime_tuning_policy get_policy(int cc, cccl_type_info accumulator_type)
   auto [_, block_size, items_per_thread, vector_load_length] = find_tuning(cc, chain);
 
   // Implement part of MemBoundScaling
-  items_per_thread = cuda::std::clamp(items_per_thread * 4 / accumulator_type.size, 1, items_per_thread * 2);
-  block_size = _CUDA_VSTD::min(block_size, (((1024 * 48) / (accumulator_type.size * items_per_thread)) + 31) / 32 * 32);
+  int candidate_ipt  = static_cast<int>(items_per_thread * 4 / accumulator_type.size);
+  items_per_thread   = cuda::std::clamp(candidate_ipt, 1, items_per_thread * 2);
+  int max_block_size = static_cast<int>((((1024 * 48) / (accumulator_type.size * items_per_thread)) + 31) / 32 * 32);
```

Comment:
nit
```suggestion
  auto candidate_ipt  = static_cast<int>(items_per_thread * 4 / accumulator_type.size);
  items_per_thread   = cuda::std::clamp(candidate_ipt, 1, items_per_thread * 2);
  auto max_block_size = static_cast<int>((((1024 * 48) / (accumulator_type.size * items_per_thread)) + 31) / 32 * 32);
```
Maybe we should also avoid hard-coding `int` by replacing them with `decltype(items_per_thread` and `decltype(block_size)`, respectively?

Reply:
(same for changes elsewhere)

Reply:
But `int` is shorter than `auto` :)

Reply:
Yeah for all CUDA-supported platforms int = int32_t, but if we were to be pedantic then the latter is preferred, then its longer than auto 😆